fdbclient/BackupContainer.actor.cpp

```diff
@@ -1305,12 +1341,15 @@ class BackupContainerFileSystem : public IBackupContainer {
 			restorable.snapshot = snapshot.get();
 			restorable.targetVersion = targetVersion;
 
-			std::vector<RangeFile> ranges = wait(bc->readKeyspaceSnapshot(snapshot.get()));
-			restorable.ranges = ranges;
+			std::pair<std::vector<RangeFile>, std::map<std::string, KeyRange>> results =
+			    wait(bc->readKeyspaceSnapshot(snapshot.get()));
+			restorable.ranges = std::move(results.first);
+			restorable.keyRanges = std::move(results.second);
```

Comment:
Can we have a sanity check in simulation for the `restorable.keyRanges`?
It can be done by getting the [beginKey, endKey) in each range file and compare it with the value in `restorable.keyRanges`.
The function to get the [beginKey, endKey) is already in `insertRangeVersion()` (https://github.com/apple/foundationdb/pull/2970/files#diff-c862e8665223bc2aa5cbb923dec24db9R717).

It helps to catch errors early, in case there is error in the metadata.

Reply:
Done.